ng-dev/auth/shared/ng-dev-token.ts

```diff
@@ -129,3 +149,49 @@ function decrypt(text: string) {
   const decrypted = Buffer.concat([decipher.update(encryptedText), decipher.final()]);
   return decrypted.toString();
 }
+
+/**
+ * Configure the AuthorizedGitClient using a temporary token from the ng-dev credential service.
+ * The token is valid for the life of the socket being open, which is expected to be for the life
+ * of the command running.
+ */
+export function configureAuthorizedGitClientWithTemporaryToken() {
+  return new Promise<void>(async (resolve, reject) => {
+    await restoreNgTokenFromDiskIfValid();
```

Comment:
Can we just assert that the caller needs to load the token first? I think we are calling this restoring multiple times now